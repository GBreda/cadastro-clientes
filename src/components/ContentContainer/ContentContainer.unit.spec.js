import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import ContentContainer from './ContentContainer.vue';

const mountContentContainer = ({ slots } = {}) => {
  return mount(ContentContainer, {
    props: {
      title: 'tÃ­tulo'
    },
    slots: {
      ...slots
    }
  });
};

describe('ContentContainer', () => {
  let wrapper;
  it('should mount the component', () => {
    wrapper = mountContentContainer();

    expect(wrapper.vm).toBeDefined();
  });

  it('should match the snapshot', () => {
    wrapper = mountContentContainer();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render default slot content', () => {
    const wrapper = mountContentContainer({
      slots: {
        default: '<div data-test-id="default">Custom Default Content</div>'
      }
    });

    expect(wrapper.find('[data-test-id="default"]').text()).toBe('Custom Default Content');
  });
});
