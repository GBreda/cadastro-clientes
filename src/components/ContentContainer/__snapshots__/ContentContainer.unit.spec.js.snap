// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentContainer > should match the snapshot 1`] = `
<div
  class="content-container"
>
  <h1
    class="pb-4"
  >
    t√≠tulo
  </h1>
  
  
</div>
`;
